src/flow_solver/flow_solver_cases/periodic_entropy_tests.cpp

```diff
@@ -0,0 +1,304 @@
+            dealii::Tensor<1,dim,std::vector<double>> ref_gradient_basis_fns_times_soln;
+            for(int idim=0; idim<dim; idim++){
+                ref_gradient_basis_fns_times_soln[idim].resize(n_quad_pts);
+                soln_grad_at_q_vect[istate][idim].resize(n_quad_pts);
+            }
+            // Apply gradient of reference basis functions on the solution at volume cubature nodes.
+            soln_basis.gradient_matrix_vector_mult_1D(soln_coeff[istate], ref_gradient_basis_fns_times_soln,
+                                                      soln_basis.oneD_vol_operator,
+                                                      soln_basis.oneD_grad_operator);
+            // Transform the reference gradient into a physical gradient operator.
+            for(int idim=0; idim<dim; idim++){
+                for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+                    for(int jdim=0; jdim<dim; jdim++){
+                        //transform into the physical gradient
+                        soln_grad_at_q_vect[istate][idim][iquad] += metric_oper.metric_cofactor_vol[idim][jdim][iquad]
+                                                                  * ref_gradient_basis_fns_times_soln[jdim][iquad]
+                                                                  / metric_oper.det_Jac_vol[iquad];
+                    }
+                }
+            }
+        }
+
+        // Loop over quadrature nodes, compute quantities to be integrated, and integrate them.
+        for (unsigned int iquad=0; iquad<n_quad_pts; ++iquad) {
+
+            std::array<double,nstate> soln_at_q;
+            std::array<dealii::Tensor<1,dim,double>,nstate> soln_grad_at_q;
+            // Extract solution and gradient in a way that the physics ca n use them.
+            for(int istate=0; istate<nstate; istate++){
+                soln_at_q[istate] = soln_at_q_vect[istate][iquad];
+                for(int idim=0; idim<dim; idim++){
+                    soln_grad_at_q[istate][idim] = soln_grad_at_q_vect[istate][idim][iquad];
+                }
+            }
+            
+            //#####################################################################
+            // Compute integrated quantities here
+            //#####################################################################
+            if (quantity == IntegratedQuantityEnum::kinetic_energy) { 
+                const double KE_integrand = this->euler_physics->compute_kinetic_energy_from_conservative_solution(soln_at_q);
+                integrated_quantity += KE_integrand * quad_weights[iquad] * metric_oper.det_Jac_vol[iquad];
+            } else if (quantity == IntegratedQuantityEnum::numerical_entropy) {
+                const double quadrature_entropy = this->euler_physics->compute_numerical_entropy_function(soln_at_q);
+                if (isnan(quadrature_entropy))  this->pcout << "WARNING: NaN entropy detected at a node!"  << std::endl;
```

Comment:
Should be std::cout instead of pcout because if the cell is not owned by processor #0, processor #0 will not go below `cell->is_locally_owned()`.

Reply:
True, that's a really good point which I hadn't thought of. Changing this.